src/main/scala/viper/gobra/frontend/Config.scala

```diff
@@ -359,8 +359,30 @@ class ScallopGobraConfig(arguments: Seq[String], isInputOptional: Boolean = fals
   )
 
   banner(
-    s""" Usage: ${GoVerifier.name} -i <input-files> [OPTIONS] OR
-       |  ${GoVerifier.name} -p <directory> [OPTIONS]
+    s""" ${GoVerifier.name} supports three modes for specifying input files that should be verified:
+       |  ${GoVerifier.name} -i <input-files> [OPTIONS] OR
+       |  ${GoVerifier.name} -p <directories> <optional project root> [OPTIONS] OR
+       |  ${GoVerifier.name} -r <optional project root> <optional include and exclude package names> [OPTIONS]
+       |
+       | Mode 1 (-i):
+       |  The first way interprets the list of files as one package that should be verified.
+       |  Optionally, positional information can be provided for each file, e.g. <path to file>@42,111, such that only
+       |  members at these positions will be verified.
+       |
+       | Mode 2 (-p):
+       |  Interprets each provided directory as a directory storing files belonging to a single package.
+       |  Verifies these packages. The project root (by default the current working directory) is used to derive a
+       |  unique package identifier, since package names might not be unique.
+       |
+       | Mode 3 (-r):
+       |  Transitively locates source files in subdirectories relative to the project root (by default the current
+       |  working directory) and groups them to packages based on the relative path and package name.
+       |  --includePackages <package names> and --excludePackages <package names> can be used to allow-
+       |  or block-list packages.
+       |
+       | Note that --include <directories> is unrelated to the modes above and controls how ${GoVerifier.name} resolves
+       | package imports.
+       |
```

Comment:
It seems to me that mode 2 and 3 are a bit confusing. Mode 2 could take a list of packages and then verify them. Mode 3 could take a list of dictonaries and verify all packages found therein (where one can still blacklist packages). The project root is a separate entity that just influences how package names are resolved. Then we would have an option to verify single files, single packages, and entire projects. Mode 2 could maybe even also use -i.

Reply:
Isn't that what we have?
I think of these modes as follows:
- mode 1: take these files and verify them, don't care about where they are located in the filesystem
- mode 2: take these packages (specified via the directory in which they are located) and verify them
- mode 3: take this module / repo / ... and verify the packages in there (or only a subset as specified via includePackages and excludePackages)

Reply:
I think the confusion comes from the concrete flags and mode 3. For mode 3, the repo is given in a roundabout way through `--projectRoot` which otherwise has a different purpose. The `--includePackages` is counterintuitive to me because it specifies additional packages in a command that is supposed to verify all packages in folders; "Verify all packages in my project, ... and also these ones". If mode 3 takes a list of dictonaries, then the `--includePackages` becomes redundant (and maybe mode 2 also becomes redundant). Regarding the flags, I think `--dictonaries` for packages and `--recursive` for something that currently does not have a provided target (like `-d /wireguard -r`) is a bit counterintuitive.

Reply:
I agree that the current separation between mode 2 and mode 3 is confusing, but at least the current description describes accurately what these modes do. From my perspective, `--recursive` should be an option for mode two, instead of a separate option (this way, we might be able to deprecate `--projectRoot`).